scripts/bld

```diff
@@ -69,8 +69,30 @@ if [ -n "${tiledb}"  ]; then
   printf "Build with TileDB: ${tiledb}\n"
   extra_opts+=" -DFORCE_BUILD_TILEDB=OFF"
   export TileDB_DIR="${tiledb}"
-  export LD_LIBRARY_PATH="${tiledb}"
-  export DYLD_LIBRARY_PATH="${tiledb}"
+
+  # It's more elegant to use
+  #   if [[ -v LD_LIBRARY_PATH ]]; then
+  #     ...
+  #   fi
+  # -- however, this is supproted by bash not sh. And this script can
+  # be invoked by python via its shell-out which seems to use sh.
+  # It's simplest to just pause set -u.
+
+  set +u
+
+  if [ -z $"LD_LIBRARY_PATH" ]; then
```

Comment:
I am under the impression that the linker ignored leading/trailing semi-colons, in which case you can just do:
> export LD_LIBRARY_PATH="${tiledb}:${LD_LIBRARY_PATH}"

that said, the extra code should not hurt anything...

Reply:
@bkmartinjr I'd rather not count on that; who knows what system's linker somewhere is non-compliant -- safety is best -- thank you though!

Reply:
`$"LD_LIBRARY_PATH"` evaluates to the string `LD_LIBRARY_PATH`; the `$` needs to be inside the quotes:

```bash
docker run --rm --entrypoint bash bash -c 'foo=1; echo $"foo" "$foo"'
# foo 1
```

`[ -z $"LD_LIBRARY_PATH" ]` is always false, so we've inadvertently been doing what @bkmartinjr suggested (always prepending `${tiledb}:`). I think this is also why the `set +u` seemed necessary; the "append" block was running even with `LD_LIBRARY_PATH` unset, which `set -u` was flagging as an error.

A common way of doing these two `if` blocks is:
```bash
export LD_LIBRARY_PATH="$tiledb${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}"
export DYLD_LIBRARY_PATH="$tiledb${DYLD_LIBRARY_PATH:+:$DYLD_LIBRARY_PATH}"
```
cf. [SO](https://stackoverflow.com/a/46892074). This also works in `sh`:
```sh
docker run --rm -it --entrypoint sh ubuntu -c 'foo=; echo "prepended_val${foo:+:$foo}"'
# prepended_val
docker run --rm -it --entrypoint sh ubuntu -c 'foo=val; echo "prepended_val${foo:+:$foo}"'
# prepended_val:val
```
though I'd think this file's shebang line ensures we're always running `bash`.

I will send a PR with a fix, as well as other changes recommended by [shellcheck](https://www.shellcheck.net/) (which flagged this issue to me).

Reply:
Thank you @ryan-williams -- `$"` is very clearly wrong and I'm absolutely appalled that I missed a basic shell-syntax error on reviewing this :(